app/src/main/java/com/hwang/xsighting/MainActivity.java

```diff
@@ -14,113 +19,193 @@
+//      showName();
+      Task<GetTokenResult> token = user.getIdToken(false);
+      Log.i(TAG, user.toString());
+    } else {
+      // No user signed in, direct them to Login
+      Log.i(TAG, "About to launch sign in");
+      // Choose authentication providers
+      List<AuthUI.IdpConfig> providers = Arrays.asList(
+              new AuthUI.IdpConfig.EmailBuilder().build());
+
+      // Create and launch sign-in intent
+      startActivityForResult(
+              AuthUI.getInstance()
+                      .createSignInIntentBuilder()
+                      .setAvailableProviders(providers)
+                      .build(),
+              RC_SIGN_IN);
+
+      Log.i(TAG, "Sign up intent Sent");
     }
+    setContentView(R.layout.activity_main);
+    BottomNavigationView bottomNavigationView = (BottomNavigationView)
+            findViewById(R.id.navigation);
+    updateRecyclerView();
+    bottomNavigationView.setOnNavigationItemSelectedListener(
+            new BottomNavigationView.OnNavigationItemSelectedListener() {
+              @Override
+              public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+                switch (item.getItemId()) {
+                  case R.id.navigation_home:
+                      Intent homeIntent = new Intent(MainActivity.this, MainActivity.class);
+                      startActivity(homeIntent);
+                  case R.id.navigation_add_sighting:
+                    Intent addSighting = new Intent(MainActivity.this, CreateSighting.class);
+                    startActivity(addSighting);
+                }
+                return true;
+              }
+            });
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-
-        if (requestCode == RC_SIGN_IN) {
-            IdpResponse response = IdpResponse.fromResultIntent(data);
-
-            if (resultCode == RESULT_OK) {
-                // Successfully signed in
-                createNewUserIfUserDoesNotExist(FirebaseAuth.getInstance().getUid());
-                showName();
-            } else {
-                // Sign in failed. If response is null the user canceled the
-                // sign-in flow using the back button. Otherwise check
-                // response.getError().getErrorCode() and handle the error.
-                // ...
-            }
-        }
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+
+    if (requestCode == RC_SIGN_IN) {
+      IdpResponse response = IdpResponse.fromResultIntent(data);
+
+      if (resultCode == RESULT_OK) {
+        // Successfully signed in
+        createNewUserIfUserDoesNotExist(FirebaseAuth.getInstance().getUid());
+//        showName();
+      } else {
+        // Sign in failed. If response is null the user canceled the

```

Comment:
You don't have any actual error handling code here; what do you want to do if the user didn't sign in?